# exploratory data analysis dalam data Student Assessment OULAD dataset

keterangan :
    This file contains the results of students’ assessments. If the student does not submit the assessment, no result is recorded. The final exam submissions is missing, if the result of the assessments is not stored in the system. This file contains the following columns:

    id_assessment – the identification number of the assessment.
    id_student – a unique identification number for the student.
    date_submitted – the date of student submission, measured as the number of days since the start of the module presentation.
    is_banked – a status flag indicating that the assessment result has been transferred from a previous presentation.
    score – the student’s score in this assessment. The range is from 0 to 100. The score lower than 40 is interpreted as Fail. The marks are in the range from 0 to 100.


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os

In [3]:
def Load_File(path) :
    if os.path.isfile(path) :
        data = pd.read_csv(path)
        data = data.replace('?', np.nan)
        print("\n" + "banyaknya baris adalah %s" % len(data))
        print("banyaknya kolom adalah %s" % len(data.columns) + "\n")
        print("tipe data :- ")
        print(data.dtypes)
        print("Missing value (null):- ")
        print(data.isnull().sum())
        
        return data
    else:
        print(path + " tidak ada")

In [4]:
df = Load_File(r'C:\Users\Kouma\Datasets\studentAssessment.csv')


banyaknya baris adalah 173912
banyaknya kolom adalah 5

tipe data :- 
id_assessment       int64
id_student          int64
date_submitted      int64
is_banked           int64
score             float64
dtype: object
Missing value (null):- 
id_assessment       0
id_student          0
date_submitted      0
is_banked           0
score             173
dtype: int64


In [5]:
df.head()

id_assessment  id_student  date_submitted  is_banked  score
0           1752       11391              18          0   78.0
1           1752       28400              22          0   70.0
2           1752       31604              17          0   72.0
3           1752       32885              26          0   69.0
4           1752       38053              19          0   79.0

# preprocessing
terdapat 173 data null pada column score yang dapat di drop

In [7]:
df = df.dropna()

menampilkan total data setelah menghilangkan missing value

In [9]:
df[5:]

id_assessment  id_student  date_submitted  is_banked  score
5                1752       45462              20          0   70.0
6                1752       45642              18          0   72.0
7                1752       52130              19          0   72.0
8                1752       53025               9          0   71.0
9                1752       57506              18          0   68.0
...               ...         ...             ...        ...    ...
173907          37443      527538             227          0   60.0
173908          37443      534672             229          0  100.0
173909          37443      546286             215          0   80.0
173910          37443      546724             230          0  100.0
173911          37443      558486             224          0   80.0

[173734 rows x 5 columns]